nixos/modules/services/networking/corerad.nix

```diff
@@ -0,0 +1,54 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.corerad;
+  configFile = pkgs.writeText "corerad.toml" cfg.config;
+in {
+  meta = {
+    maintainers = with maintainers; [ mdlayher ];
+  };
+
+  options.services.corerad = {
+    enable = mkEnableOption "CoreRAD IPv6 NDP RA daemon";
+
+    config = mkOption {
+      default = "";
+      example = ''
+        [[interfaces]]
+        name = "eth0"
+        send_advertisements = true
+      '';
+      type = types.lines;
```

Comment:
You might consider making this an attribute set as per https://github.com/Infinisil/rfcs/blob/config-option/rfcs/0042-config-option.md for more flexibility.

Reply:
This seems really neat. I don't suppose there is an example anywhere of transforming these settings into a TOML configuration file? I would like to do this properly, but unfortunately I am pretty new to Nix and don't understand a lot of the intracacies of the language as of yet. An example would help greatly.

Reply:
I'd bet @Infinisil has something pretty awesome he could share.

Reply:
although good to keep in mind. That rfc still isn't fully ratified or implemented, I think it's good as-is given the current paradigms/tooling.

Reply:
Is an empty configuration acceptable?

Reply:
No, you have to explicitly configure at least one interface to be used to serve router advertisements.

Reply:
I would suggest removing the default value which will force the user enter a value.

Reply:
TIL thanks! I will update this today.